modules/gplazma2-jython/src/main/java/org/dcache/gplazma/jython/JythonPlugin.java

```diff
@@ -0,0 +1,175 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JythonPlugin.class);
+
+    private final PythonInterpreter PI = new PythonInterpreter();
+    private List<Path> pluginPathList = new ArrayList<>();
+
+    public JythonPlugin(Properties properties) {
+        // constructor
+        try {
+            pluginPathList = listDirectoryContents("gplazma2-jython");  // TODO fix this, currently just reads from local folder
+        } catch (IOException e) {
+            LOGGER.error("Error reading Jython directory");
+        }
+    }
+
+    private static List<Path> listDirectoryContents(String directoryPath) throws IOException {
+        List<Path> elements = new ArrayList<>();
+        Path path = Paths.get(directoryPath);
+
+        // Check if the path is a directory
+        if (Files.isDirectory(path)) {
+            // Use DirectoryStream to iterate through the directory
+            try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(path)) {
+                for (Path p : directoryStream) {
+                    elements.add(p);
+                }
+            }
+        } else {
+            throw new IllegalArgumentException("Path is not a directory: " + directoryPath);
+        }
+
+        return elements;
+    }
+
+    public void authenticate(
+            Set<Object> publicCredentials,
+            Set<Object> privateCredentials,
+            Set<Principal> principals
+    ) throws AuthenticationException {
+
+        for (Path pluginPath : pluginPathList) {
+            // for each path in pluginPathList
+            try {
+                // ===========
+                // IO
+                // ===========
+                // read file into string
+                LOGGER.debug("gplazma2-jython running " + pluginPath.toString());
+                String content = Files.lines(
+                        Paths.get(pluginPath.toString())
+                ).collect(
+                        Collectors.joining("\n")
+                );
+                // execute file content, loading the function "py_auth" into the namespace
+                PI.exec(content);
+
+                // ======================
+                // Prepare Python Data
+                // ======================
+                // prepare datatypes for python
+                PySet pyPublicCredentials = new PySet();
+                PySet pyPrivateCredentials = new PySet();
+                PySet pyPrincipals = new PySet();
+
+                // Only handle select datatypes
+                for (Object pubCred : publicCredentials) {
+                    if (
+                            pubCred instanceof String
+                            || pubCred instanceof Number // all number datatypes
+                    ) {
+                        pyPublicCredentials.add(pubCred);
+                    } else {
+                        LOGGER.warn(
+                            "GPlazma2-Jython: Public Credential "
```

Comment:
The same here, please use placeholders. That way, string concatenation will run only when debug logging is enabled.